include/jemalloc/internal/rtree_types.h

```diff
@@ -25,11 +26,17 @@ typedef struct rtree_s rtree_t;
 #define RTREE_HEIGHT_MAX						\
     ((1U << (LG_SIZEOF_PTR+3)) / RTREE_BITS_PER_LEVEL)
 
+/*
+ * Number of leafkey/leaf pairs to cache.  Each entry supports an entire leaf,
+ * so the cache hit rate is typically high even with a small number of entries.
+ * Two entries is enough to keep the hit rate high even when an arena uses a
+ * combination of dss and mmap.
+ */
+#define RTREE_CTX_NCACHE 2
```

Comment:
Should be based on cache misses - a tree search is potentially ~3 cacheline misses, so it can be CACHELINE_SIZE / 16 * 3 big before being worse? ~12?  Although since the cache is linear the prefetcher might help a bit.